image_builder/notify.py

```diff
@@ -33,19 +31,19 @@ def __init__(self, send_notifications: bool = True):
             except KeyError as e:
                 raise ValueError(f"{e} environment variable must be set")
 
-    def post_build_progress(self, progress: Progress, codebase: Codebase):
+    def post_build_progress(self, progress: Progress, text_blocks=None):
```

Comment:
I think we need a test for when `text_blocks` is `None`.

Reply:
If `text_blocks` is never meant to be anything other than a `Dict` of a set structure, could we:

* create a class with those three properties
* put any required validation/defaulting into the constructor of the class
* remove the validation above and change this line to...

```suggestion
    def post_build_progress(self, progress: Progress, text_blocks: WhateverWeCallTheClass):
```

I'm also wondering if `text_blocks` is the best name for that variable. Maybe `progress_data`?

Reply:
so we remove the tight coupling between `Notify` and `Codebase` to add a new coupling between `Notify` and `WhateverWeCallTheClass`

What I've done, In my mind at least, is one step forward towards a generic `Notify` object that takes pre-formatted slack blocks and posts them (akin to what @antroy-madetech suggests)

In truth, `ci-image-builder`'s notification object should be a `NotificationFormatter` object that pings into the `Notify` object, but it's a bit of scope creep

Reply:
Feels like there are further iterations to do on this code, but we probably have more pressing concerns right now.